### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by leeks-int**
Signed-off-by: Alexey Oralov alexey.oralov@intel.com

### CODE REVIEW

**Review by omalyshe — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- [ ] tests (change in tests)
+- [ ] infrastructure (change in infrastructure and CI)
+- [ ] documentation (documentation update)
+
+Leave empty if none of the other choices apply.
+
+## Pull request checklist
+Please check if your PR fulfills the following requirements:
+- [ ] Tests for the changes have been added (for bug fixes / features)
+- [ ] Docs have been reviewed and added / update is needed (for bug fixes / features)
```
Comment (AlexVeprev) [inline .github/pull_request_template.md:18]: This item looks confusing to me because if two different sentences

Comment (omalyshe) [reply .github/pull_request_template.md:18]: How about "Docs have been reviewed and added OR documentation update is needed"? Or it is better to divide into two bullets like 
- [ ] Documentation has been updated for this change 
- [ ] Documentation update is needed (add details)

Comment (AlexVeprev) [reply .github/pull_request_template.md:18]: Well, let's step a bit back. What is "checklist"? Per my understanding it is something that is _required_ to be completed before merge. But these documentation sentences looks _optional_ (as well as tests sentence). What do you think?

Maybe it shouldn't be called "checklist" or you may consider rephrasing like this and require them to be fully completed before merge:

- [ ] Tests are implemented or not applicable.
- [ ] Documentation is updated/planned or not applicable.

Comment (omalyshe) [reply .github/pull_request_template.md:18]: The idea is to remind a developer to add tests for new features (a must) and some bugs (when applicable). The same is for documentation - we want to make sure the documentation is updated / planned for update accordingly.

I agree with the comment regarding "checklist" name. What about the following? 

## Adding test(s) is required for changes labeled as a "new feature"
- [ ] Tests added
- [ ] Not applicable 

## New features and some bug fixes might require documentation update
- [ ] Documentation is being updated for this change (_you included documentation change in this PR or provide PR# for doc update_)
- [ ] Documentation update is needed (_add details what kind of doc changes are required_)

Comment (AlexVeprev) [reply .github/pull_request_template.md:18]: Yes, something like that. Let me try to summarize my vision:

-----

### Description 
_Comprehensive description of proposed changes_


Fixes # - _issue number(s) if exists_

### Type(s)

_Choose one or multiple, leave empty if none of the other choices apply_
_Add respective label(s) to PR if you have permissions_

- [ ] bug fix - _change which fixes an issue_
- [ ] new feature - _change which adds functionality_
- [ ] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Test(s)

- [ ] added - _required for new features and for some bug fixes_
- [ ] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [ ] not needed

### Breaks backward compatibility
- [ ] Yes
- [ ] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### Other information

Comment (omalyshe) [reply .github/pull_request_template.md:18]: Looks great for me. The only small change I'd suggest is renaming sections:
Type(s) -> Type of change
Test(s) -> Tests

Comment (leeks-int) [reply .github/pull_request_template.md]: Updated

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+### Description 
+_Add comprehensive description of proposed changes_
+
+
+Fixes # - _issue number(s) if exists_
+
+### Type of change
+
+_Choose one or multiple, leave empty if none of the other choices apply_
+_Add respective label(s) to PR if you have permissions_
```
Comment (AlexVeprev) [inline .github/pull_request_template.md:10]: Hmm...during parsing these two lines are gathered into one line, please try to add empty line between

Comment (leeks-int) [reply .github/pull_request_template.md:11]: Added

**Review by AlexVeprev — APPROVED** [review]

**Review by omalyshe — APPROVED** [review]

**Pull request merged**